src/lib/peers/PeerManager.ts

```diff
@@ -0,0 +1,237 @@
+import os from 'os'
+import fs from 'fs'
+import net from 'net'
+import tls from 'tls'
+import Path from 'path'
+import KeyManager from "@polykey/keys/KeyManager";
+import PeerInfo, { Address } from "@polykey/peers/PeerInfo";
+import { firstPromiseFulfilled } from '@polykey/utils';
+import MulticastBroadcaster from "@polykey/peers/MulticastBroadcaster";
+import RPCMessage from '@polykey/rpc/RPCMessage';
+import createX509Certificate from '@polykey/pki/PublicKeyInfrastructure'
+
+interface SocialDiscovery {
+  // Must return a public pgp key
+  name: string
+  findUser(handle: string, service: string): Promise<string>
+}
+
+const keybaseLookup = async (handle: string, service: string): Promise<string> => {
```

Comment:
Should eventually be abstracted into a strategy pattern. Not all the time we will be using keybase.

Reply:
There is a strategy pattern just above this function in the form of an interface for `SocialDiscovery`. The keybase was set as a default `SocialDiscovery` but it is left to the user to add more.

Reply:
I was expecting a directory with different strategies in different files. But it's cool for now.